### PR TITLE
[full-ci] Fix several file handling issues for apps

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
This PR fixes a bunch of scenarios where mediaviewer (and possibly other apps) were broken:
- `getUrlForResource` was using an ugly hack to retrieve a webdav url before `webDavPath` was introduced in https://github.com/owncloud/web/pull/6287, we could now get rid of that hack and make this function work for spaces and a bunch of other contexts work as well (Kudos for that PR!!)
- mediaviewer: clicking next/previous updated the url with the resource `path` instead of the `webDavPath`: that broke reloading
- router.parseQuery: our `qs.parse` invocation was missing the `allowDots` option, so nested objects in query were not correctly parsed and closing the apps, at least on public links was broken 

## Related Issues
There are two oCIS bugs left, which cause the mediaviewer to fail
- for images in spaces https://github.com/owncloud/ocis/issues/3088 
- for shares https://github.com/owncloud/ocis/issues/2462

With oC 10 *everything* seems to work just fine.

Biggest remaining issue is, that reloading the mediaviewer will only load the correct context folder for personal files or public links - i.e. if you open the mediaviewer in the shares overview you can cycle through your shared pictures, if you reload the page you will cycle through the folder from which the file was shared.
Fixing that is way beyond the scope of this PR and needs to be handled in a follow up PR.

## Motivation and Context
It's annoying when the mediaviewer or other apps don't work in all contexts :-P 

## How Has This Been Tested?
Apart from the before mentioned issues, I'm not aware of any remaining ones.
With these changes the mediaviewer works in `Personal`,  `Favorites`, `Shared With Others`, `Shared Via Link` and `Spaces` views.

~~Just `Shared With Me` I haven't tested yet.~~ `Shared With Me` was tested, works.

~~Query Parsing seems to behave a little weird, needs to be double checked again.~~
~~I needed the change to fix closing apps on public links, but now reloading seems broken in `Personal`. Marking the PR as WIP for now.~~
Seems to have been a hiccup, cannot reproduce anymore.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by kulmann — COMMENTED** [review]
Looks really promising already! 😍

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23237/66/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L94.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles3</strong> https://drone.owncloud.com/owncloud/web/23243/57/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUILogin-openidLogin_feature-L22.png
  

### webUILogin-openidLogin_feature-L29.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23243/66/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:188

File: packages/web-app-files/src/helpers/resources.js [hunk 1]
```diff
@@ -177,11 +177,11 @@ export function buildSpace(space) {
 }
 
 export function buildWebDavFilesPath(userId, path) {
-  return `/files/${userId}/${path}`
+  return '/' + `files/${userId}/${path}`.split('/').filter(Boolean).join('/')
 }
 
 export function buildWebDavSpacesPath(spaceId, path) {
-  return `/spaces/${spaceId}/${path}`
+  return '/' + `spaces/${spaceId}/${path}`.split('/').filter(Boolean).join('/')
```
Comment (dschmidt) [inline packages/web-app-files/src/helpers/resources.js:184]: Looking at this now it's pretty useless to put the `/` in a separate string to concatenate it :-D

File: changelog/unreleased/bugfix-app-file-handling [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Bugfix: File handling in apps
+
+We fixed loading and saving files in apps in all contexts. It's now to open files in apps in personal files, favorites, share views and spaces.
```
Comment (pascalwengerter) [inline changelog/unreleased/bugfix-app-file-handling:3]:
```suggestion
We fixed loading and saving files in apps in all contexts. It's now possible to open files in apps in personal files, favorites, share views and spaces.
```

File: changelog/unreleased/bugfix-app-file-handling [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+https://github.com/owncloud/web/pull/6456/files
```
Comment (pascalwengerter) [inline changelog/unreleased/bugfix-app-file-handling:5]:
```suggestion
https://github.com/owncloud/web/pull/6456
```

File: packages/web-app-files/src/helpers/resources.js (hunk 1 above)
Comment (dschmidt) [reply packages/web-app-files/src/helpers/resources.js:184]: Ah, nope - it's not. The split/join afterwards removes the leading `/`

**Review by pascalwengerter — COMMENTED** [review]
Looking good so far, two minor comments on the changelog and waiting for more incoming changes (?)

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23267/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

**Review by pascalwengerter — APPROVED** [review]
LGTM

**Pull request merged**